js/pro/binance.js

```diff
@@ -550,15 +550,15 @@ module.exports = class binance extends binanceRest {
         }
         const id = this.safeString2 (trade, 't', 'a');
         const timestamp = this.safeInteger (trade, 'T');
-        const price = this.safeFloat2 (trade, 'L', 'p');
-        let amount = this.safeFloat (trade, 'q');
+        const price = this.safeString2 (trade, 'L', 'p');
```

Comment:
@samgermain please next time, test it in python as well, it would be very easy to spot that the cost is returned as a string there because this method does not use `safeTrade` 
```
CCXT v2.7.23
binanceusdm.watchMyTrades(LTC/USDT:USDT)
[{'amount': '0.100',
  'cost': '9.593', <-------
  'datetime': '2023-02-01T11:38:15.834Z',
  'fee': {'cost': 0.0038372, 'currency': 'USDT'},
  'id': '640401834',
  'info': {'L': '95.93',
           'N': 'USDT',
```